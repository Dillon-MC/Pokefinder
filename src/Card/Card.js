import React from 'react';
import { useEffect, useState } from 'react';
const Card = ({
    name='',
    type='',
    weight='',
    hiddenAbility='',
    baseStats=[],
    sprite='',
    Expand=false,
    extendCardCallback }) => {
    const [CardStyles, setCardStyles] = useState({});
    const [cardButton, setCardButton] = useState({});
    const [infoTextStyle, setInfoTextStyle] = useState(null);
    const textColors = ['green','blue','darkPurple','darkBlue','purple','yellow'];

    function changeLayout() {
        if(!Expand) {
            defualtCard();
        } else {
            expandCard();
        }
    }

    // Sets the card styles to defualt
    function defualtCard() {
        setCardContentStyles('Card', 'pokemonIcon-ShiftCenter', 'extendCardButton', 'More..');
        setInfoTextStyle(
            <div className='ma1 infotext textShadow'>
                <h4>{`Type: ${type}`}</h4>
                <h4>{`Weight: ${weight}kg`}</h4>
                <h4>{`HA: ${hiddenAbility}`}</h4>
            </div>
        );
    }

    // Sets the card styles to expanded mode so more info can be displayed.
    function expandCard() {
        setCardContentStyles('Card-Expand', 'pokemonIcon-ShiftLeft', 'extendCardButton', '');
        setInfoTextStyle(null);

        setTimeout(() => {
            setCardButton({cardButtonStyle: 'extendCardButton extendCardButton-Expand', cardButtonText: 'Less..'});
            let stats = baseStats.map((stat, index) => 
                <h4 className={textColors[index]} key={index}>{stat.stat.name}: {stat.base_stat}</h4>);

                setInfoTextStyle(
                    <div className="infoText-Expand textShadow">
                        <h4>{`Type: ${type}`}</h4>
                        <h4>{`Weight: ${weight}kg`}</h4>
                        <h4>{`HA: ${hiddenAbility}`}</h4>
                        {stats}
                    </div>);
        }, 350);
    }

    function setCardContentStyles(cardClass, cardIcon, cardButtonClass, cardButtonText) {
        setCardStyles({Card: cardClass, Icon: cardIcon});
        setCardButton({cardButtonStyle: cardButtonClass, cardButtonText: cardButtonText});
    }

    useEffect(changeLayout,[Expand, baseStats]);

    return (
        <div className={`tc br ba bw2 br3 dib Enlarge pa3 shadow-5 f4 redwhiteGradient ${CardStyles.Card}`}>
            <div>
                <div className={CardStyles.Icon}>
                    <h4 className="ma0 mb3">{name}</h4>
                    <img className='pokemonIcon'
                        src={sprite}
                        alt="icon" 
                    />
                </div>
                {infoTextStyle}
                <button onClick={extendCardCallback} className={cardButton.cardButtonStyle}>{cardButton.cardButtonText}</button>
            </div>
        </div>
    );
}

export default Card;