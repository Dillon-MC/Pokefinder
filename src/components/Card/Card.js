import React from 'react';
import { useEffect, useState } from 'react';
const Card = ({
    name='',
    type='',
    weight='',
    hiddenAbility='',
    baseStats=[],
    sprite='',
    Expand=false,
    extendCardCallback }) => {
    const [CardStyles, setCardStyles] = useState({});
    const [cardButton, setCardButton] = useState({});
    const [infoTextStyle, setInfoTextStyle] = useState(null);
    const textColors = ['green','blue','darkPurple','darkBlue','purple','yellow'];

    function changeLayout() {
        if(!Expand) {
            setInfoTextStyle(
                <p className="ma1 infoText">
                    {`Type: ${type}`}
                    <br/>
                    {`Weight: ${weight}kg`}
                    <br/>
                    {`HA: ${hiddenAbility}`}
                </p>
            );
            setCardStyles({Card: 'Card', Icon: 'pokemonIcon-ShiftCenter'});
            setCardButton({cardButtonStyle: 'extendCardButton', cardButtonText: 'More..'});
        } else {
            setInfoTextStyle(null);
            setCardStyles({Card: 'Card-Expand', Icon: 'pokemonIcon-ShiftLeft'});
            setCardButton({cardButtonStyle: 'extendCardButton', cardButtonText: ''});

            setTimeout(() => {
                setCardButton({cardButtonStyle: 'extendCardButton extendCardButton-Expand', cardButtonText: 'Less..'});
                let stats = baseStats.map((stat, index) => 
                <h4 className={textColors[index]} key={index}>{stat.stat.name}: {stat.base_stat}</h4>);

                setInfoTextStyle(
                    <div className="infoText-Expand textShadow">
                        <h4>{`Type: ${type}`}</h4>
                        <h4>{`Weight: ${weight}kg`}</h4>
                        <h4>{`HA: ${hiddenAbility}`}</h4>
                        {stats}
                    </div>);
            }, 300);
        }
    }

    useEffect(changeLayout,[Expand, baseStats]);

    return (
        <div className={`tc br ba bw2 br3 dib Enlarge pa3 shadow-5 f4 redwhiteGradient ${CardStyles.Card}`}>
            <div>
                <div className={CardStyles.Icon}>
                    <h4 className="ma0 mb3">{name}</h4>
                    <img className='pokemonIcon'
                        src={sprite}
                        alt="icon" 
                    />
                </div>
                {infoTextStyle}
                <button onClick={extendCardCallback} className={cardButton.cardButtonStyle}>{cardButton.cardButtonText}</button>
            </div>
        </div>
    );
}

export default Card;