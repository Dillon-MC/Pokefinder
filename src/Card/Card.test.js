import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import Card from './Card';

configure({ adapter: new Adapter() });

it('expect Card component to match snapshot', () => {
   expect(shallow(<Card />)).toMatchSnapshot();
});