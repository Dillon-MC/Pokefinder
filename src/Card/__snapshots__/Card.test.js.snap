// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect Card component to match snapshot 1`] = `
<div
  className="tc br ba bw2 br3 dib Enlarge pa3 shadow-5 f4 redwhiteGradient undefined"
>
  <div>
    <div>
      <h4
        className="ma0 mb3"
      />
      <img
        alt="icon"
        className="pokemonIcon"
        src=""
      />
    </div>
    <button />
  </div>
</div>
`;
